nomos-services/key-management-system/src/backend/mod.rs

```diff
@@ -0,0 +1,20 @@
+use crate::KMSOperator;
+use bytes::Bytes;
+use either::Either;
+use overwatch_rs::DynError;
+
+#[async_trait::async_trait]
+pub trait KMSBackend {
+    type SupportedKeys;
+    type KeyId;
+    type Settings;
+
+    fn new(settings: Self::Settings) -> Self;
+    fn register(
+        &mut self,
+        key: Either<Self::SupportedKeys, Self::KeyId>,
```

Comment:
Question: If the `key` is `Either::Right(key_id)`, this function loads the key from its storage?

Reply:
Well, it depends on the backend. The idea is that all this needs to be reproducible. So if some system had a key, the node goes down then it is recovered, it should continue operations with the same key right?

Reply:
Yes. Hmm, then I'm thinking about whether we really need `Either::Right(key_id)`. When a node is restarted, the node can load all keys from its storage, so that the subsequent operations (e.g. `sign`, etc), can succeed.

Reply:
If we have `Either::Right(key_id)` in this API, and if the backend loads a key only when the user calls `register`, then the user must keep all `key_id`s that he needs in the storage, so that he can call the `register` manually when the node is restarted.

Reply:
> If we have `Either::Right(key_id)` in this API, and if the backend loads a key only when the user calls `register`, then the user must keep all `key_id`s that he needs in the storage, so that he can call the `register` manually when the node is restarted. If this is the intended behavior, I'm okay with having `Either::Right(key_id)`.

Yes, this is the intended behaviour. Also they may be loaded or generated. We do not know yet.